pkg/controllers/selection/controller.go

```diff
@@ -39,17 +39,19 @@ const controllerName = "selection"
 
 // Controller for the resource
 type Controller struct {
-	kubeClient   client.Client
-	provisioners *provisioning.Controller
-	preferences  *Preferences
+	kubeClient     client.Client
+	provisioners   *provisioning.Controller
+	preferences    *Preferences
+	volumeTopology *VolumeTopology
 }
 
 // NewController constructs a controller instance
 func NewController(kubeClient client.Client, provisioners *provisioning.Controller) *Controller {
 	return &Controller{
-		kubeClient:   kubeClient,
-		provisioners: provisioners,
-		preferences:  NewPreferences(),
+		kubeClient:     kubeClient,
+		provisioners:   provisioners,
+		preferences:    NewPreferences(),
+		volumeTopology: NewVolumeTopology(kubeClient),
```

Comment:
Should we wrap VolumeTopology into Preferences and not expose it to the Controller?

Reply:
I think these are fully separate concepts. Preferences is all about relaxing preferred node affinities. Volume topology is a fully separate scheduling plugin.

Reply:
That said, as we continue to build out scheduling features, I'm always game for a reimagining of the implementation to something more streamlined -- there just wasn't anything obvious to me as I've built it piece by piece.

Reply:
> Preferences is all about relaxing preferred node affinities.

Aaah okay, that makes sense - on first reading I thought Preferences was the way we're getting all the scheduling logic into the controller. I suppose it was just first. 

Agreed though, it might be nice to have some kind of uniform contract for all our scheduler plugins to implement and unifying all of them into a different wrapper maybe?